lib/persistent-knex-model-mixin.js

```diff
@@ -0,0 +1,170 @@
+'use strict'
+const _ = require('lodash')
+
+module.exports = function PersistentBookshelfModelMixin (Model, knex, inOptions, userOptions) {
+  if (typeof Model.convertFromPersistent !== 'function') {
+    Model.convertFromPersistent = function (data) {
+      return data
+    }
+  }
+
+  if (typeof Model.convertToPersistent !== 'function') {
+    Model.convertToPersistent = function (data) {
+      return data
+    }
+  }
+
+  /**
+   * Set data from database.
+   *
+   * Applies filters for conversion from database format to normal format.
+   */
+  Model.prototype.setDataPersistent = function (data) {
+    this.setData(this._applyFilter(this.constructor.convertFromPersistent, data))
+  }
+
+  Model.prototype.getDataPersistent = function () {
+    return this._applyFilter(this.constructor.convertToPersistent, this.getData())
+  }
+
+  Model.prototype.save = function * (options) {
+    let myKnex = knexOrTransaction(knex, options)
+    return myKnex.update(this.getDataPersistent()).into(Model.tableName).where('id', this.id).then()
+  }
+
+  // Delete row based on where options
+  Model.prototype.destroy = function (selectOptions) {
+    return knex(Model.tableName).where(selectOptions).del()
+  }
+
+  Model.fromDatabaseModel = function (data) {
+    if (!data) return null
+    // If result is an array (return from knex select query), take the first object
+    if (Array.isArray(data)) {
+      data = data[0]
+    }
+    const model = new Model()
+    model.setDataPersistent(data)
+    return model
+  }
+
+  Model.fromDataPersistent = function (data) {
+    const model = new Model()
+    model.setDataPersistent(data)
+    return model
+  }
+
+  Model.findAll = function * findAll () {
+    // this returns an array of JSON objects
+    return knex(Model.tableName).select().then()
+  }
+  Model.findById = function * findById (id, options) {
+    let myKnex = knexOrTransaction(knex, options)
+    return myKnex.from(Model.tableName).select().where('id', id).then(fromDatabaseModel)
+  }
+  /*
+  Model.findOne = function findOne (options) {
+    return DbModel.findOne(options).then(fromDatabaseModel)
+  }
+
+  for (let method of ['count', 'findAndCount', 'max', 'min', 'sum', 'truncate']) {
+    Model[method] = DbModel[method].bind(DbModel)
+  }
+
+  Model.build = function build (values, options) {
+    const model = Model.fromData(values)
+
+    return DbModel.build(model.getDataPersistent(), options)
+  }
+  */
+
+  // Optional: transaction
+  // set this to knex transaction if you want this operation to be done inside transaction.
+  Model.create = function * create (values, options) {
+    const model = Model.fromData(values)
+    const modelPersistent = model.getDataPersistent()
+    let myKnex = knexOrTransaction(knex, options)
```

Comment:
I left these here because these functionalities existed in the original persistent-model-mixin.  I haven't implemented these because Notary doesn't need them, but these serve as reminders we may want to implement them in the future, e.g., if Ledger needs them.